libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.kt

```diff
@@ -0,0 +1,142 @@
+package com.mapbox.services.android.navigation.v5.utils
+
+import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.core.utils.TextUtils
+import com.mapbox.geojson.Point
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgressState
+import java.util.ArrayList
+import java.util.Arrays
+import java.util.Comparator
+
+class RouteUtils {
+
+    /**
+     * Looks at the current [RouteProgressState] and returns if
+     * is [RouteProgressState.ROUTE_ARRIVED].
+     *
+     * @param routeProgress the current route progress
+     * @return true if in arrival state, false if not
+     */
+    fun isArrivalEvent(routeProgress: RouteProgress): Boolean {
+        val currentState = routeProgress.currentState()
+        return currentState != null && currentState == RouteProgressState.ROUTE_ARRIVED
+    }
+
+    /**
+     * Looks at the current [RouteProgress] list of legs and
+     * checks if the current leg is the last leg.
+     *
+     * @param routeProgress the current route progress
+     * @return true if last leg, false if not
+     * @since 0.8.0
+     */
+    fun isLastLeg(routeProgress: RouteProgress): Boolean {
+        val legs = routeProgress.directionsRoute().legs()
+        val currentLeg = routeProgress.currentLeg()
+        legs?.let { legsList ->
```

Comment:
Should we make it like:
```suggestion
        return legs?.let { legsList ->
            currentLeg == legsList[legsList.size - 1]
        } ?: false
```